Remotion/Validation/Core/Validators/ExactLengthValidator.cs

```diff
@@ -15,18 +15,72 @@
 // along with re-motion; if not, see http://www.gnu.org/licenses.
 // 
 using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
 using JetBrains.Annotations;
+using Remotion.FunctionalProgramming;
+using Remotion.Utilities;
 using Remotion.Validation.Implementation;
+using Remotion.Validation.Results;
 
 namespace Remotion.Validation.Validators
 {
-  public class ExactLengthValidator : LengthValidator
+  public class ExactLengthValidator : ILengthValidator
   {
+    int ILengthValidator.Min => Length;
+
+    int? ILengthValidator.Max => Length;
+
+    public int Length { get; }
+
+    public string ErrorMessage { get; }
+
+    public ValidationMessage ValidationMessage { get; }
+
     public ExactLengthValidator (int length, [NotNull] ValidationMessage validationMessage)
-        : base(length, length, $"The value must have exactly {length} characters.", validationMessage)
     {
+      ArgumentUtility.CheckNotNull(nameof(validationMessage), validationMessage);
+
+      if (length < 0)
+        throw new ArgumentOutOfRangeException("length", "Value cannot be less than zero.");
```

Comment:
This will be updated in the follow-up ticket mentioned in MaximumLengthValidator (length must be breater than zero and message for updated argument check in follow-up ticket: "Value must be be greater than zero.")